### PR TITLE
mds: misc multimds fixes part2

### PR DESCRIPTION
**Pull request by ukernel**

### CODE REVIEW

**Comment by ukernel** [comment]
some commits are in https://github.com/ceph/ceph/pull/12274. The first new commit is "mds: remove stale warning in log replay code"

File: src/mds/StrayManager.cc [hunk 1]
```diff
@@ -771,20 +771,20 @@ void StrayManager::eval_remote_stray(CDentry *stray_dn, CDentry *remote_dn)
...
-    // NOTE: we repeat this check in _rename(), since our submission path is racey.
-    if (!remote_dn->is_projected()) {
-      if (remote_dn->is_auth() && remote_dn->dir->can_auth_pin()) {
-        reintegrate_stray(stray_dn, remote_dn);
-      } else if (!remote_dn->is_auth() && stray_dn->is_auth()) {
-        migrate_stray(stray_dn, remote_dn->authority().first);
-      } else {
-        dout(20) << __func__ << ": not reintegrating" << dendl;
-      }
+  // NOTE: we repeat this check in _rename(), since our submission path is racey.
```
Comment (xiexingguo) [inline src/mds/StrayManager.cc]: s/racey/racy/

**Comment by jcsp** [comment]
On a test branch with these patches, I'm seeing failures in TestStrays.test_migration_on_shutdown and TestFailover.test_grow_shrink -- possibly something here is breaking rank stopping.

**Comment by ukernel** [comment]
It's caused by unrelated bug. fixed by	"mds: fix MDCache::trim() with max == 0"

**Comment by jcsp** [comment]
Oh yeah, I happened to also fix that on my purgequeue PR https://github.com/ceph/ceph/pull/12786/commits/7f6187ae780ef9760a286049dc7c97fd888fbd2f

**Pull request merged**